integrated_channels/moodle/client.py

```diff
@@ -36,9 +36,14 @@ def inner(self, *args, **kwargs):
             # This cannot be converted to JSON thus the above fails miserably.
             # The above can fail with different errors depending on the format of the returned page.
             # Moodle of course does not tell us what is wrong in any part of this HTML.
-            raise ClientError('Moodle API task "{method}" failed due to unknown error.'.format(
-                method=method.__name__), response.status_code
-            ) from error
+            log_format_keys = {'method': method.__name__,
+                               'code': response.status_code,
+                               'enterprise_configuration_id': self.enterprise_configuration.id}
+            log_format_string = '''\
+                                Moodle API task "{method}" for enterprise_configuration "{enterprise_configuration_id}"
+                                failed due to unknown error with code "{code}".
+                                '''
+            raise ClientError(log_format_string.format(**log_format_keys), response.status_code) from error
```

Comment:
Love this logging!

Reply:
I think this looks great, and works perfectly well- but in general we're trying to trend towards f strings instead of `format` and I think it might make things a little clearer here. Feel free to format this a bit more but my thoughts are something along the lines of 
```
       log_format_string = (f'Moodle API task "{method.__name__}" for enterprise_configuration "{self.enterprise_configuration.id}}" '
                             f'failed due to unknown error with code "{response.status_code}".')
```

Reply:
I'm also curious if config ID is the best identifier here- obviously that field does exist in the table, but perhaps a more helpful way to easily know which configuration customer would be to log `enterprise_customer_id`. Maybe we could do both? IDK, what are your thoughts? Only because we don't do much with the config IDs and they're not readily available unless you're querying read replica

Reply:
I have recently been using that `generate_formatted_log` as much as possible, sometimes we can't fill in all the info but mostly it works great, and makes logs consistent too. and the `message` part of that I now use f`` strings yes generally